packages/rendermime/src/widgets.ts

```diff
@@ -114,23 +121,31 @@ class RenderedHTML extends RenderedHTMLCommon {
     }
     Private.appendHtml(this.node, source);
     if (options.resolver) {
-      this._urlResolved = Private.handleUrls(this.node, options.resolver,
-                                             options.linkHandler);
+      Private.handleUrls(this.node, options.resolver, options.linkHandler).then(() => {
+        this._ready.resolve(undefined);
```

Comment:
I guess this type of logic is the reason the ready promise exists in the first place?